gb-clock/examples/capped.rs

```diff
@@ -0,0 +1,98 @@
+use simplelog::{ColorChoice, Config, LevelFilter, TermLogger, TerminalMode};
+
+use gb_bus::Bus;
+use gb_clock::{Clock, Tick, Ticker};
+
+use std::time::{Duration, Instant};
+
+#[derive(Default)]
+struct FakeBus {}
+
+impl Bus<u8> for FakeBus {
+    fn read(&self, _adr: u16) -> Result<u8, gb_bus::Error> {
+        Ok(0xff)
+    }
+    fn write(&mut self, _adr: u16, _data: u8) -> Result<(), gb_bus::Error> {
+        Ok(())
+    }
+}
+impl Bus<u16> for FakeBus {
+    fn read(&self, _adr: u16) -> Result<u16, gb_bus::Error> {
+        Ok(0xffff)
+    }
+    fn write(&mut self, _adr: u16, _data: u16) -> Result<(), gb_bus::Error> {
+        Ok(())
+    }
+}
+
+struct FakeCPU {
+    pub tick_count: usize,
+}
+
+impl Ticker<FakeBus> for FakeCPU {
+    fn cycle_count(&self) -> Tick {
+        Tick::TCycle
+    }
+
+    fn tick(&mut self, _adr_bus: &mut FakeBus) {
+        self.tick_count += 1;
+    }
+}
+
+struct FakePPU {
+    pub tick_count: usize,
+}
+
+impl Ticker<FakeBus> for FakePPU {
+    fn cycle_count(&self) -> Tick {
+        Tick::MCycle
+    }
+
+    fn tick(&mut self, _adr_bus: &mut FakeBus) {
+        self.tick_count += 1;
+    }
+}
+
+fn main() {
+    TermLogger::init(
+        LevelFilter::Info,
+        Config::default(),
+        TerminalMode::Mixed,
+        ColorChoice::Auto,
+    )
+    .expect("cannot setup terminal logger");
+
+    let mut cpu = FakeCPU { tick_count: 0 };
+    let mut ppu = FakePPU { tick_count: 0 };
+    let mut bus = FakeBus::default();
+    let mut clock = Clock::default();
+    let one_sec = Duration::from_millis(1_000);
```

Comment:
Why `from_millis(1_000` and not `from_secs(1)` ?

Reply:
I started by outputing at a higher frequency, before settling to one second, but yes it can be simplified.